src/app/data-model/DecodableList.h

```diff
@@ -134,12 +147,24 @@ class DecodableList
     Iterator begin() const { return Iterator(mReader); }
 
     /*
-     * Compute the size of the list.  This can fail if the TLV is malformed.  If
+     * Compute the size of the list. This can fail if the TLV is malformed. If
      * this succeeds, that does not guarantee that the individual items can be
      * successfully decoded; consumers should check Iterator::GetStatus() when
-     * actually decoding them.
+     * actually decoding them. If there is no list (null valued list case) then
+     * the size is considered to be zero.
      */
-    CHIP_ERROR ComputeSize(size_t * size) const { return mReader.CountRemainingInContainer(size); }
+    CHIP_ERROR ComputeSize(size_t * size) const
+    {
+        if (mReader.GetContainerType() == TLV::kTLVType_NotSpecified)
+        {
+            *size = 0;
+            return CHIP_NO_ERROR;
```

Comment:
That said, this part we might need for a default-initialized DecodableList to represent an empty list, and the same for the `Next` change....

Reply:
Yeah this might be handy for the default-initialized DecodableList as having better behaviour. You can also reset a DecodableList to this (using ClearReader) if that's desirable.